addons/onboarding/static/src/views/form/onboarding_step_form_controller.js

```diff
@@ -20,15 +20,20 @@ export default class OnboardingStepFormController extends FormController {
     async save({ closable, ...otherParams }) {
         const saved = await super.save(otherParams);
         if (saved) {
-            const { reloadOnFirstValidation, reloadAlways } = this.stepConfig;
-            const validationResponse = await this.orm.call(
-                'onboarding.onboarding.step',
-                'action_validate_step',
-                [this.stepName],
-            );
-            if (reloadAlways || (reloadOnFirstValidation && validationResponse === "JUST_DONE")) {
-                this.action.restore(this.action.currentController.jsId);
-            } else if (closable) {
+            let shouldClose = closable;
+            if (await this.stepIsValidationConditionMet) {
```

Comment:
Looking at this again, I'm not a fan of naming this something related to "valid".
As not being valid would imply some form validation and showing the user he has not completed required fields for example, while here we just close the window normally.

Maybe "isStepCompleted" then?

Reply:
Yes of course

Reply:
It could be `stepCanBeValidated`, I think "completed" could be confusing because it's not validated, while this function will trigger validation if this value is true.

Reply:
@awa-odoo What do you think sir ? i'm ok with both

Reply:
@flch-odoo Well if you look at what "action_validate_step" does, it marks the step as "done".

Globally (and especially in JS) the concept of "validation" usually implies validating form inputs and the like to mark them as invalid if not filled properly.

So I still vote for "isStepCompleted".
(It's very much a detail and preference though, I understand your point, both could be argued to be "better").

Reply:
My point was that the step isn't "done"/"completed" yet. The work is done, but the step isn't.

Reply:
In this case it means (to me) that the form fields from this step are "completed", as in "filled", not that the step is "finished".

But we're arguing semantics at this point 😄